Document/0x05e-Testing-Cryptography.md

```diff
@@ -132,6 +150,8 @@ KeyPair keyPair = keyPairGenerator.generateKeyPair();
 
 This sample creates the RSA key pair with a key size of 4096-bit (i.e. modulus size).
 
+Note: there is a widespread fasle believe that the NDK should be used to hide cryptographic operations and hardcoded keys. However, using this mechanisms is not effective. Attackers can still use tools to find the mechanism used and make dumps of the key in memory. Next, the control flow can be analyzed with IDA(pro). From Android Nougat onward, it is not allowed to use private APIs, instead: public APIs need to be called, which further impacts the effectiveness of hiding it away as described in the [Android Developers Blog](https://android-developers.googleblog.com/2016/06/android-changes-for-ndk-developers.html "Android changes for NDK developers")
```

Comment:
fasle -> false 

However, using this mechanisms is not effective. -> However, using such "Security through Obscurity" mechanisms is not effective.

Attackers can still use tools to find the mechanism used and make dumps of the key in memory.
-> Attackers can still use simple tools as `strings` to find hardcoded secrets in shared objects or make dumps of the key in memory while using the app.